site/docs/v1/tech/lambdas/index.adoc

```diff
@@ -45,11 +54,98 @@ function populate(jwt, user, registration) {
 }
 ----
 
-Each lambda documentation page will have an example lambda implementation.
+=== Making an HTTP Request
+
+include::docs/v1/tech/shared/_advanced-edition-blurb.adoc[]
+
+Here is an example of a FusionAuth lambda that adds additional claims to a JWT based on an HTTP request:
+
+[source,javascript]
+----
+function populate(jwt, user, registration) {
+  var response = fetch("https://api.example.com/api/status?"+user.id, {
+    method: "GET",
+    headers: {
+      "Content-Type": "application/json"
+    }
+  });
+  
+  if (response.status === 200) {
+    var jsonResponse = JSON.parse(response.body);
+    jwt.status = jsonResponse.status;
+  } else {
+    jwt.status = "basic";
+  }
+----
+
+==== Headers
+
+You can provider request header values in a number of different ways:
+
+[source,javascript,title="An anonymous object"]
+----
+headers: {
+  "Content-Type": "application/json"
+}
+----
+
+[source,javascript,title="A hash or map"]
+----
+headers: new Headers({
+   "Content-Type": "application/json"
+})
+----
+
+[source,javascript,title="An array"]
+----
+headers: new Headers([
+    ["Content-Type", "application/json"]
+])
+----
+
+==== Response 
+
+A response object will be returned. It will have the following fields:
+
+[.api]
+[field]#headers# [type]#[Array]#::
+The headers returned by the response.
```

Comment:
The headers will actually be an object (map) of header name to value where you may have one or more header values by a single key. 

For example:

```
var headerValue = response.headers['Content-Type']; 
```

Reply:
Hmmm. I did some testing and it appears that we only get back the last header.

That is if I have this PHP function

```
<?php

header('X-test: def');
header('X-test: abc');

header('Accept: text/html');
header('Accept: application/xhtml+xml');

?>

{"quote": "abc"}
```

I'll only get the last value for each header in the object. `abc` for the `X-test` header and `application/xhtml+xml` for the `Accept` header.

Reply:
Also, all the headers are shifted to lower-case: 

```
{"accept":"application/xhtml+xml","connection":"close","content-type":"text/html; charset=UTF-8","date":"Mon, 14 Mar 2022 22:02:07 GMT","host":"host.docker.internal:8000","x-powered-by":"PHP/7.3.29-to-be-removed-in-future-macOS","x-test":"abc"}
```

Reply:
> header('Accept: text/html');

Yes, but that is a map, not an array, or maybe I am mis-understanding. Assuming the data type would be `[type]#[Object]#::` or however we normally indicate that the returned value is a key based object. 

If there were more than one value the value would back as a comma separated value I believe. Not all headers names are allowed to have more than one value, that is up to the spec / browser, etc.